remote.go

```diff
@@ -316,8 +318,14 @@ func certificateCheckCallback(_cert *C.git_cert, _valid C.int, _host *C.char, da
 		cert.Kind = CertificateHostkey
 		ccert := (*C.git_cert_hostkey)(unsafe.Pointer(_cert))
 		cert.Hostkey.Kind = HostkeyKind(ccert._type)
-		C.memcpy(unsafe.Pointer(&cert.Hostkey.HashMD5[0]), unsafe.Pointer(&ccert.hash_md5[0]), C.size_t(len(cert.Hostkey.HashMD5)))
-		C.memcpy(unsafe.Pointer(&cert.Hostkey.HashSHA1[0]), unsafe.Pointer(&ccert.hash_sha1[0]), C.size_t(len(cert.Hostkey.HashSHA1)))
+		switch cert.Hostkey.Kind {
```

Comment:
In my testing they don't come back zero-initialized for some reason. I'm not sure why that is?
Running libgit2 1.1.0-1 on arch linux

Reply:
ah then that means that libgit2 was successfully able to ask libssh2 for that particular certificate hash and it _should_ be the correct hash.

unless, of course, there's a bug lurking somewhere :S

Reply:
Ah, I get it now. It will actually return all the hashes, which is why it's a bit field. The returned hashes didn't match my expectations from the output of `ssh-keyscan -t rsa whoami.filippo.io | ssh-keygen -l -E md5 -f -` but it's likely me that is holding it wrong. I'll do some more testing, and remove this part of the PR.

Reply:
Nevermind, I was authenticating with Ed25519 when I thought I was using RSA :facepalm:, which is why the hash'es didn't match.